Mathlib/CategoryTheory/Bicategory/Adjunction.lean

```diff
@@ -292,6 +292,67 @@ end Equivalence
 
 end
 
+noncomputable
+section
+
+/-- A class giving a chosen right adjoint of a 1-morphism `left`. -/
+structure RightAdjoint (left : a ⟶ b) where
+  /-- The right adjoint to `left`. -/
+  right : b ⟶ a
+  /-- The adjunction between `left` and `right`. -/
+  adj : left ⊣ right
+
+/-- The existence of a right adjoint of `f`. -/
+class IsLeftAdjoint (left : a ⟶ b) : Prop where  mk' ::
+  nonempty : Nonempty (RightAdjoint left)
+
+theorem IsLeftAdjoint.mk (adj : f ⊣ g) : IsLeftAdjoint f :=
+  ⟨⟨g, adj⟩⟩
+
+/-- Use the axiom of choice to extract a right adjoint from an `IsLeftAdjoint` instance. -/
+def getRightAdjoint (f : a ⟶ b) [IsLeftAdjoint f] : RightAdjoint f :=
+  Classical.choice IsLeftAdjoint.nonempty
+
+/-- The right adjoint of a 1-morphism. -/
+def rightAdjoint (f : a ⟶ b) [IsLeftAdjoint f] : b ⟶ a :=
+  (getRightAdjoint f).right
+
+@[inherit_doc] scoped postfix:max "⁺⁺" => rightAdjoint
```

Comment:
Are these notions for the left and right adjoint standard in the literature?

Reply:
Regarding this, it doesn't seem that this is used anywhere else. It might be better to just write it as `right|Adjoint f`.